src/nwb_conversion_tools/datainterfaces/ecephys/spikeglx/spikeglxdatainterface.py

```diff
@@ -1,29 +1,47 @@
-"""Authors: Cody Baker and Ben Dichter."""
+"""Authors: Cody Baker, Heberto Mayorquin and Ben Dichter."""
 from datetime import datetime
 from pathlib import Path
 from typing import Optional
 
-from spikeextractors import SpikeGLXRecordingExtractor, SubRecordingExtractor, RecordingExtractor
+import spikeextractors as se
+from spikeinterface import BaseRecording
+from spikeinterface.extractors import SpikeGLXRecordingExtractor
+from spikeinterface.core.old_api_utils import OldToNewRecording
+
+
+from spikeextractors import SubRecordingExtractor, RecordingExtractor
 from pynwb.ecephys import ElectricalSeries
 
 from ..baserecordingextractorinterface import BaseRecordingExtractorInterface
 from ..baselfpextractorinterface import BaseLFPExtractorInterface
 from ....utils import get_schema_from_method_signature, get_schema_from_hdmf_class, FilePathType
 
 
-def fetch_spikeglx_metadata(file_path: FilePathType, recording: RecordingExtractor, metadata: dict):
-    file_path = Path(file_path)
-    session_id = file_path.parent.stem
+def fetch_spikeglx_metadata(recording: BaseRecording, metadata: dict):
+    """
+    Fetches the session_start_time from the meta readings and adds it to the metadata.
+    """
 
-    if isinstance(recording, SubRecordingExtractor):
-        n_shanks = int(recording._parent_recording._meta.get("snsShankMap", [1, 1])[1])
-        session_start_time = datetime.fromisoformat(recording._parent_recording._meta["fileCreateTime"]).astimezone()
+    # Support for old spikeextractors objects
+    recording = recording._kwargs.get("oldapi_recording_extractor", recording)
+    if isinstance(recording, RecordingExtractor):
+        if isinstance(recording, SubRecordingExtractor):
+            meta = recording._parent_recording._meta
```

Comment:
This leading underscore is marking `meta` as private. It means that external libraries are not advised to rely on this variable and that spikeinterface is not guaranteeing backwards compatibility support going forward. If could be OK as a temporary fix, but ideally I'd rather not access private attributes from other repos.

Reply:
This I inherited from the code before and it is not my solution.

 That said, It is private attribute in the old `spikeextractors` but they are not moving that anymore. As you can see in the new `spikeinterface` (below that) the same meta dictionary (which is just a the metadata file parsed) is provided as well but now with a public interface. If you want to remove the access in `spikeextractors` we can parse the metadata ourselves but that seems like an overkill.
 
 What do you think?

Reply:
I think it's fine if this is just used for the old extractor and a public attribute for the new extractor